File: solutions/cpp/darts/darts.cpp
#include "darts.h"

namespace darts {

    auto score(double x, double y) -> int
    {
        double r = sqrt(x*x + y*y);

        if (r > 10)
            return 0;
        else if (r > 5)
            return 1;
        else if (r > 1)
            return 5;
        else
            return 10;
    }

}  // namespace darts

File: solutions/csharp/darts/Darts.cs
public static class Darts
{
    public static double DistanceFromCenter(double x, double y)
    {
        return Math.Sqrt(x * x + y * y);
    }
    public static int Score(double x, double y)
    {
        switch (DistanceFromCenter(x, y))
        {
            case <= 1:
                return 10;
            case <= 5:
                return 5;
            case <= 10:
                return 1;
            default:
                return 0;
        }
    }
}
